### PR TITLE
mcs: reduce duplicated definitions 

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Ref #5836.

### What is changed and how does it work?

This PR inits cluster ID for tso and resource manager server and move some common definition to another package.

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/mcs/tso/server/server.go [hunk 1]
```diff
@@ -540,12 +536,9 @@ func (s *Server) startGRPCAndHTTPServers(l net.Listener) {
 }
 
 func (s *Server) startServer() (err error) {
-	// TODO: uncomment the following code to generate a unique cluster id from the given tsoClusterIDPath
-	// after we add rpc for the client to retrieve the clusgter id from the server then use it in every
-	// request for verification.
-	// if s.clusterID, err = etcdutil.InitClusterID(s.etcdClient, tsoClusterIDPath); err != nil {
-	//	return err
-	// }
+	if s.clusterID, err = etcdutil.GetClusterID(s.etcdClient, utils.ClusterIDPath); err != nil {
```
Comment (binshi-bing) [inline pkg/mcs/tso/server/server.go:539]: See my TODO: comment on the left side. We can't enable cluster id initialization for tso service for now -- every rpc will check if the cluster id in the request header is equal to the one in the server. For pd, when the pd client starts, it calls GetMemebers to get the cluster id, but tso service/client hasn't defined the equivalent rpc to retrieve the cluster id from the server and use it in the requests thereafter. We need a separate pr which combines this change in tso server, the change in tso protobuf and the tso client initialization. @rleungx

**Review by binshi-bing — CHANGES_REQUESTED** [review]

File: pkg/mcs/tso/server/server.go (hunk 1 above)
Comment (binshi-bing) [reply pkg/mcs/tso/server/server.go:539]: @rleungx , how about leaving the code as what it is here? I'll make the change mentioned above after you merge this pr.

Comment (rleungx) [reply pkg/mcs/tso/server/server.go]: done

**Review by lhy1024 — APPROVED** [review]

**Review by JmPotato — APPROVED** [review]

**Pull request merged**